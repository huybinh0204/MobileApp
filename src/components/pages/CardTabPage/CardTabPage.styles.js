import styled, { css } from '@emotion/native';
import { BaseText, IconSvg, Link, MainButton } from '_components/atoms';
import Spacing from '_constants/Spacing';

export const DebitCardBanner = styled(IconSvg)(({ theme }) => ({
  alignSelf: 'center',
  marginVertical: theme.spacing.s,
}));

export const DebitCardDisclaimer = styled(BaseText)(({ theme }) => ({
  ...theme.typography.legal,
  color: theme.colors.beta500,
  marginBottom: theme.spacing.s,
  marginHorizontal: theme.spacing.m,
  textAlign: 'center',
}));

export const BottomSheetContainer = styled.View(({ theme }) => ({
  paddingHorizontal: theme.spacing.m,
}));

export const BottomSheetTitle = styled(BaseText)(({ theme }) => ({
  ...theme.typography.heading5,
  color: theme.colors.beta900,
  marginBottom: theme.spacing.xs,
}));

export const BottomSheetContent = styled(BaseText)(({ theme }) => ({
  ...theme.typography.mainContentRegular,
  color: theme.colors.beta500,
  marginBottom: theme.spacing.s,
}));

export const SupportNumberLink = styled(Link)(({ theme }) => ({
  color: theme.colors.delta500,
  marginBottom: theme.spacing.s,
}));

export const BottomSheetButton = styled(MainButton)(({ theme }) => ({
  marginTop: theme.spacing.s,
}));

export const listStyles = css({
  paddingHorizontal: Spacing.m,
});
