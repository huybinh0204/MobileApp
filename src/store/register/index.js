export * from './register_reducer';
export * from './register_sagas';
