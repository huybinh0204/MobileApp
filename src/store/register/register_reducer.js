import { createActions, createReducer } from 'reduxsauce';
import Immutable from 'seamless-immutable';

const { Types, Creators } = createActions(
  {
    trackEvent: ['event', 'eventType', 'properties'],
    trackEventSuccess: ['message'],
    trackEventFails: ['message'],
  },
  {
    prefix: '@@BE-REGISTER/',
  }
);

export const RegisterActions = Creators;
export const RegisterTypes = Types;

// INITIAL STATE
export const INITIAL_STATE = Immutable({});

// ACTION HANDLERS
const trackEventSuccess = (state, { message }) => {
  console.info('Track Event SUCCESS:', message);
  return state;
};

const trackEventFails = (state, { message }) => {
  console.warn('Track Event FAIL:', message);
  return state;
};

// REDUCER
export const RegisterReducer = createReducer(INITIAL_STATE, {
  [Types.TRACK_EVENT_SUCCESS]: trackEventSuccess,
  [Types.TRACK_EVENT_FAILS]: trackEventFails,
});
