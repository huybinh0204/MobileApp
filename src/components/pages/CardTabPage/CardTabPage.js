import { useFocusEffect, useNavigation } from '@react-navigation/native';
import React, { useCallback, useMemo, useState } from 'react';
import ENV from 'react-native-config';
import { useDispatch, useSelector } from 'react-redux';
import { BottomSheet, IconSvg } from '_components/atoms';
import { PressableList } from '_components/molecules';
import { MainScreenLayout } from '_components/organisms';
import { EVENTS, EVENT_TYPES, FEATURE_FLAGS, ICONS, LINK_TYPES, NAVIGATION } from '_constants';
import { useBooleanFeatureFlag } from '_hooks';
import strings from '_localization/index';
import { AccountActions, AccountSelectors } from '_store/account';
import { RegisterActions } from '_store/register';
import { openExternalLink } from '_utilities/ExternalLinks';
import { normalize } from '_utilities/screen';
import {
  BottomSheetButton,
  BottomSheetContainer,
  BottomSheetContent,
  BottomSheetTitle,
  DebitCardBanner,
  DebitCardDisclaimer,
  listStyles,
  SupportNumberLink,
} from './CardTabPage.styles';

const itemIcon = (icon) => (props) => <IconSvg icon={icon} {...props} />;

const CardTabPage = () => {
  const dispatch = useDispatch();
  const { navigate } = useNavigation();

  const accountExternalId = useSelector(AccountSelectors.getExternalId);
  const hasActiveDebitCards = useSelector(AccountSelectors.hasActiveDebitCards);
  const hasDebitCardWaitingToActivate = useSelector(AccountSelectors.hasDebitCardWaitingToActivate);
  const isChangePinEnabled = useBooleanFeatureFlag(FEATURE_FLAGS.CHANGE_PIN);

  const [bottomSheet, setBottomSheet] = useState({ isVisible: false, title: '', content: '' });

  const pressableItems = useMemo(() => {
    return [
      {
        hide: !hasDebitCardWaitingToActivate,
        testID: 'ActivateCard',
        title: strings.cardTab_list_activate_card,
        leftItem: itemIcon(ICONS.activatedCard),
        rightItem: itemIcon(ICONS.arrowRight),
        onPress() {
          navigate(NAVIGATION.card.activateCard);
        },
      },
      {
        hide: !hasActiveDebitCards,
        testID: 'ChangePIN',
        title: strings.cardTab_list_change_pin,
        leftItem: itemIcon(ICONS.changePin),
        rightItem: itemIcon(ICONS.arrowRight),
        onPress() {
          isChangePinEnabled
            ? navigate(NAVIGATION.card.setDebitCardPinInfo, { isChangePinVariant: true })
            : setBottomSheet({
                isVisible: true,
                title: strings.card.changePinInfo.header,
                content: strings.card.changePinInfo.changePinMessage,
              });
        },
      },
      {
        title: strings.cardTab_list_lock_card,
        leftItem: itemIcon(ICONS.lockCard),
        rightItem: itemIcon(ICONS.arrowRight),
        onPress() {
          setBottomSheet({
            isVisible: true,
            title: strings.cardTab_lockCard_title,
            content: strings.cardTab_lockCard_content,
          });
          dispatch(RegisterActions.trackEvent(EVENTS.LOCK_CARD_OPENED, EVENT_TYPES.TRACK));
        },
      },
      {
        title: strings.cardTab_list_replace_card,
        leftItem: itemIcon(ICONS.replaceCard),
        rightItem: itemIcon(ICONS.arrowRight),
        onPress() {
          setBottomSheet({
            isVisible: true,
            title: strings.cardTab_replaceCard_title,
            content: strings.cardTab_replaceCard_content,
          });
          dispatch(RegisterActions.trackEvent(EVENTS.REPLACE_CARD_OPENED, EVENT_TYPES.TRACK));
        },
      },
      {
        title: strings.cardTab_list_find_atm,
        leftItem: itemIcon(ICONS.atm),
        rightItem: itemIcon(ICONS.arrowRight),
        async onPress() {
          await openExternalLink(ENV.FIND_ATM_URI);
          dispatch(RegisterActions.trackEvent(EVENTS.FIND_ATM_OPENED, EVENT_TYPES.TRACK));
        },
      },
      {
        title: strings.cardTab_list_find_place_to_deposit_cash,
        leftItem: itemIcon(ICONS.findDepositCash),
        rightItem: itemIcon(ICONS.arrowRight),
        async onPress() {
          await openExternalLink(ENV.FIND_PLACE_TO_DEPOSIT_CASH_URI);
          dispatch(
            RegisterActions.trackEvent(EVENTS.FIND_PLACE_TO_DEPOSIT_CASH_OPENED, EVENT_TYPES.TRACK)
          );
        },
      },
    ];
  }, [dispatch, navigate, hasActiveDebitCards, hasDebitCardWaitingToActivate, isChangePinEnabled]);

  const closeBottomSheet = () => {
    setBottomSheet({ ...bottomSheet, isVisible: false });
  };

  const fetchData = useCallback(() => {
    dispatch(AccountActions.fetchKinlyDebitCards(accountExternalId));
  }, [accountExternalId, dispatch]);

  useFocusEffect(fetchData);

  return (
    <MainScreenLayout testID="CardTabPage" title={strings.cardTab_title}>
      <DebitCardBanner
        icon={ICONS.debitCardBanner}
        height={normalize(150)}
        width={normalize(230)}
        testID="CardBanner"
      />
      <DebitCardDisclaimer>{strings.cardTab_disclaimer}</DebitCardDisclaimer>
      <PressableList
        contentContainerStyle={listStyles}
        items={pressableItems}
        testID="CardTabList"
      />
      <BottomSheet
        closeOnDragDown
        isVisible={bottomSheet.isVisible}
        onClose={closeBottomSheet}
        height={normalize(280)}
      >
        <BottomSheetContainer>
          <BottomSheetTitle>{bottomSheet.title}</BottomSheetTitle>
          <BottomSheetContent>{bottomSheet.content}</BottomSheetContent>
          <SupportNumberLink href={strings.kinly_support_number} type={LINK_TYPES.PHONE}>
            {strings.kinly_support_number}
          </SupportNumberLink>
          <BottomSheetButton onPress={closeBottomSheet}>
            {strings.cardTab_bottomSheet_continue}
          </BottomSheetButton>
        </BottomSheetContainer>
      </BottomSheet>
    </MainScreenLayout>
  );
};

export default CardTabPage;
