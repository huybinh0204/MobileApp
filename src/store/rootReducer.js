import { combineReducers } from 'redux';
import actionWatchReducer from 'redux-action-watch/lib/reducer';
import { resettableReducer } from 'reduxsauce';
import { AccountReducer } from './account';
import { AuthenticationReducer } from './authentication';
import { CustomerReducer } from './customer';
import { FundDirectDepositStoreReducer } from './pageStore/fundDirectDepositStore';
import { RegisterReducer } from './register';
import { TransferReducer } from './transfer';

const resettable = resettableReducer('RESET');

export default combineReducers({
  account: resettable(AccountReducer),
  auth: resettable(AuthenticationReducer),
  customer: resettable(CustomerReducer),
  fundDirectDepositStoreReducer: resettable(FundDirectDepositStoreReducer),
  register: RegisterReducer,
  transfer: resettable(TransferReducer),
  watcher: actionWatchReducer, // Must be last
});
