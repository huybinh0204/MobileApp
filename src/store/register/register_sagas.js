import { call, put, select } from 'redux-saga/effects';
import { EVENT_TYPES } from '_constants';
import { CustomerSelectors } from '_store/customer';
import { RegisterActions } from './register_reducer';

export function* trackEvent(api, action) {
  const { event, properties = {}, eventType = EVENT_TYPES.SCREEN } = action;
  const customerExternalId = yield select(CustomerSelectors.getExternalId);
  const userId = customerExternalId ?? 'unknown';

  if (!EVENT_TYPES[eventType]) {
    console.warn(`Invalid argument: ${eventType} is not a valid eventType`);
  }

  try {
    yield call(api.sendEvent, event, properties, eventType, userId);
    yield put(RegisterActions.trackEventSuccess(event));
  } catch (e) {
    yield put(RegisterActions.trackEventFails(`${event} with ${e}`));
  }
}
