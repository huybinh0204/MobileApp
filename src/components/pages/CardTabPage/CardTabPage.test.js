import { useNavigation } from '@react-navigation/native';
import * as Split from '@splitsoftware/splitio-react';
import { fireEvent, render, waitFor } from '@testing-library/react-native';
import React from 'react';
import { Linking } from 'react-native';
import ENV from 'react-native-config';
import { useDispatch } from 'react-redux';
import { createStore } from 'redux';
import { DEBIT_CARD_STATUS, EVENTS, EVENT_TYPES, FEATURE_FLAGS, NAVIGATION } from '_constants';
import strings from '_localization/index';
import { RegisterActions } from '_store/register';
import rootReducer from '_store/rootReducer';
import ComponentWithProviders from '_test/mocks/ComponentWithProviders';
import { navigationMock } from '_test/mocks/navigation';
import CardTabPage from './CardTabPage';

jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useNavigation: jest.fn(),
}));

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: jest.fn(),
}));

describe('CardTabPage', () => {
  let mockStore;
  let mockedDispatch = jest.fn();
  let mockedNavigation = navigationMock(jest.fn);

  beforeEach(() => {
    useDispatch.mockReturnValue(mockedDispatch);
    useNavigation.mockReturnValue(mockedNavigation);

    mockStore = createStore(rootReducer, {
      account: {
        kinlyDebitCards: [{ status: DEBIT_CARD_STATUS.READY_TO_ACTIVATE }],
      },
    });

    jest.spyOn(Split, 'useClient').mockImplementation(() => ({
      getTreatment(flag) {
        expect(flag).toEqual(FEATURE_FLAGS.CHANGE_PIN);
        return 'on';
      },
    }));
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match the snapshot', () => {
    const { toJSON } = render(<ComponentWithProviders component={CardTabPage} store={mockStore} />);

    expect(toJSON()).toMatchSnapshot();
  });

  it('should display the Page Title', () => {
    const { getByText } = render(
      <ComponentWithProviders component={CardTabPage} store={mockStore} />
    );

    expect(getByText('Card')).not.toBeNull();
  });

  it('should display the card banner', () => {
    const { getByTestId } = render(
      <ComponentWithProviders component={CardTabPage} store={mockStore} />
    );

    expect(getByTestId('CardBanner')).not.toBeNull();
  });

  it('should display a list of pressables with the correct number of items', () => {
    const { queryByTestId, getByTestId, getAllByTestId } = render(
      <ComponentWithProviders component={CardTabPage} store={mockStore} />
    );

    expect(queryByTestId('ActivateCard')).not.toBeNull();
    expect(getByTestId('CardTabList')).not.toBeNull();
    expect(getAllByTestId('ItemTitle')).toHaveLength(5);
  });

  it('should not have a card activation link when the account debit card has been activated', async () => {
    mockStore = createStore(rootReducer, {
      account: {
        kinlyDebitCards: [{ status: DEBIT_CARD_STATUS.ACTIVE }],
      },
    });

    const { queryByTestId } = render(
      <ComponentWithProviders component={CardTabPage} store={mockStore} />
    );

    expect(queryByTestId('ActivateCard')).toBeNull();
  });

  it('should not have a card activation nor change PIN button when there are no debit cards', () => {
    mockStore = createStore(rootReducer, {
      account: {
        kinlyDebitCards: [],
      },
    });

    const { getAllByTestId, queryByTestId } = render(
      <ComponentWithProviders component={CardTabPage} store={mockStore} />
    );

    expect(queryByTestId('ActivateCard')).toBeNull();
    expect(queryByTestId('ChangePIN')).toBeNull();
    expect(getAllByTestId('ItemTitle')).toHaveLength(4);
  });

  it('should open Find ATM link and send event', async () => {
    const linkingSpy = jest.spyOn(Linking, 'openURL');
    const trackEventSpy = jest.spyOn(RegisterActions, 'trackEvent');

    const { getByText } = render(
      <ComponentWithProviders component={CardTabPage} store={mockStore} />
    );

    const findATMButton = getByText('Find ATM');

    await waitFor(() => {
      fireEvent.press(findATMButton);
    });

    expect(linkingSpy).toBeCalledWith(ENV.FIND_ATM_URI);
    expect(trackEventSpy).toHaveBeenCalledWith(EVENTS.FIND_ATM_OPENED, EVENT_TYPES.TRACK);
  });

  it('should open Find Place to Deposit Cash in external browser and send event', async () => {
    const linkingSpy = jest.spyOn(Linking, 'openURL');
    const trackEventSpy = jest.spyOn(RegisterActions, 'trackEvent');

    const { getByText } = render(
      <ComponentWithProviders component={CardTabPage} store={mockStore} />
    );

    const depositCashButton = getByText('Find cash deposit location');

    await waitFor(() => {
      fireEvent.press(depositCashButton);
    });

    expect(linkingSpy).toBeCalledWith(ENV.FIND_PLACE_TO_DEPOSIT_CASH_URI);
    expect(trackEventSpy).toHaveBeenCalledWith(
      EVENTS.FIND_PLACE_TO_DEPOSIT_CASH_OPENED,
      EVENT_TYPES.TRACK
    );
  });

  it('should navigate to Activate Card', async () => {
    const { getByText } = render(
      <ComponentWithProviders component={CardTabPage} store={mockStore} />
    );

    const goActivateCardButton = getByText(strings.cardTab_list_activate_card);

    await waitFor(() => {
      fireEvent.press(goActivateCardButton);
    });

    expect(mockedNavigation.navigate).toBeCalledWith(NAVIGATION.card.activateCard);
  });
});
