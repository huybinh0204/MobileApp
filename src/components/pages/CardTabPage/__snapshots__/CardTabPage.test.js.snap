// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardTabPage should match the snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgb(242, 242, 242)",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <RNSScreenContainer
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          activityState={2}
          collapsable={false}
          forwardedRef={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "display": "flex",
                  "overflow": undefined,
                },
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                  "marginTop": 0,
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                needsOffscreenAlphaCompositing={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <RCTSafeAreaView
                        backgroundColor={null}
                        emulateUnlessSupported={true}
                        style={
                          Array [
                            Object {
                              "backgroundColor": "#FAFAFA",
                              "flex": 1,
                            },
                            undefined,
                          ]
                        }
                        testID="CardTabPage"
                      >
                        <View
                          style={
                            Array [
                              Object {
                                "alignItems": "center",
                                "height": 108.5,
                                "justifyContent": "center",
                              },
                              undefined,
                            ]
                          }
                        >
                          <View
                            accessible={true}
                            collapsable={false}
                            focusable={true}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "left": 36,
                                "opacity": 1,
                                "position": "absolute",
                                "zIndex": 10,
                              }
                            }
                            testID="MenuButton"
                          >
                            <SvgMock
                              height={54.5}
                              width={54.5}
                            />
                          </View>
                          <Text
                            allowFontScaling={false}
                            style={
                              Array [
                                Object {
                                  "color": "#1F1D25",
                                  "fontFamily": "Poppins-SemiBold",
                                  "fontSize": 29,
                                  "lineHeight": 40,
                                },
                                undefined,
                              ]
                            }
                          >
                            Card
                          </Text>
                        </View>
                        <View
                          onLayout={[Function]}
                          style={
                            Array [
                              Array [
                                Object {
                                  "flex": 1,
                                },
                                undefined,
                              ],
                              Object {
                                "paddingBottom": 0,
                              },
                            ]
                          }
                        >
                          <SvgMock
                            height={271.5}
                            style={
                              Array [
                                Object {
                                  "alignSelf": "center",
                                  "marginVertical": 29,
                                },
                                undefined,
                              ]
                            }
                            testID="CardBanner"
                            width={416.5}
                          />
                          <Text
                            allowFontScaling={false}
                            style={
                              Array [
                                Object {
                                  "color": "#5F6A77",
                                  "fontFamily": "Poppins",
                                  "fontSize": 20,
                                  "lineHeight": 28,
                                  "marginBottom": 29,
                                  "marginHorizontal": 58,
                                  "textAlign": "center",
                                },
                                undefined,
                              ]
                            }
                          >
                            Kinly VisaÂ® Debit Card is issued
 by The Bancorp Bank, Member FDIC.
                          </Text>
                          <RCTScrollView
                            ItemSeparatorComponent={[Function]}
                            contentContainerStyle={
                              Object {
                                "paddingHorizontal": 58,
                              }
                            }
                            data={
                              Array [
                                Object {
                                  "hide": false,
                                  "leftItem": [Function],
                                  "onPress": [Function],
                                  "rightItem": [Function],
                                  "testID": "ActivateCard",
                                  "title": "Activate card",
                                },
                                Object {
                                  "hide": true,
                                  "leftItem": [Function],
                                  "onPress": [Function],
                                  "rightItem": [Function],
                                  "testID": "ChangePIN",
                                  "title": "Change PIN",
                                },
                                Object {
                                  "leftItem": [Function],
                                  "onPress": [Function],
                                  "rightItem": [Function],
                                  "title": "Lock and unlock card",
                                },
                                Object {
                                  "leftItem": [Function],
                                  "onPress": [Function],
                                  "rightItem": [Function],
                                  "title": "Replace card",
                                },
                                Object {
                                  "leftItem": [Function],
                                  "onPress": [Function],
                                  "rightItem": [Function],
                                  "title": "Find ATM",
                                },
                                Object {
                                  "leftItem": [Function],
                                  "onPress": [Function],
                                  "rightItem": [Function],
                                  "title": "Find cash deposit location",
                                },
                              ]
                            }
                            getItem={[Function]}
                            getItemCount={[Function]}
                            keyExtractor={[Function]}
                            onContentSizeChange={[Function]}
                            onLayout={[Function]}
                            onMomentumScrollBegin={[Function]}
                            onMomentumScrollEnd={[Function]}
                            onScroll={[Function]}
                            onScrollBeginDrag={[Function]}
                            onScrollEndDrag={[Function]}
                            removeClippedSubviews={false}
                            renderItem={[Function]}
                            scrollEventThrottle={50}
                            stickyHeaderIndices={Array []}
                            testID="CardTabList"
                            viewabilityConfigCallbackPairs={Array []}
                          >
                            <View>
                              <View
                                onLayout={[Function]}
                                style={null}
                              >
                                <View
                                  accessible={true}
                                  collapsable={false}
                                  focusable={true}
                                  onClick={[Function]}
                                  onResponderGrant={[Function]}
                                  onResponderMove={[Function]}
                                  onResponderRelease={[Function]}
                                  onResponderTerminate={[Function]}
                                  onResponderTerminationRequest={[Function]}
                                  onStartShouldSetResponder={[Function]}
                                  style={
                                    Object {
                                      "alignItems": "center",
                                      "alignSelf": "stretch",
                                      "flexDirection": "row",
                                      "minHeight": 50,
                                      "opacity": 1,
                                    }
                                  }
                                  testID="ActivateCard"
                                >
                                  <SvgMock
                                    height={58}
                                    testID="LeftItem"
                                    width={58}
                                  />
                                  <Text
                                    allowFontScaling={false}
                                    marginLeft={true}
                                    style={
                                      Array [
                                        Object {
                                          "flex": 1,
                                          "fontFamily": "Poppins-SemiBold",
                                          "fontSize": 25.5,
                                          "lineHeight": 35.5,
                                          "marginLeft": 29,
                                        },
                                        null,
                                      ]
                                    }
                                    testID="ItemTitle"
                                  >
                                    Activate card
                                  </Text>
                                  <SvgMock
                                    height={45.5}
                                    testID="RightItem"
                                    width={45.5}
                                  />
                                </View>
                                <View
                                  style={
                                    Array [
                                      Object {
                                        "borderColor": "#F2F3F4",
                                        "borderWidth": 1,
                                        "height": 1,
                                        "width": "100%",
                                      },
                                      undefined,
                                    ]
                                  }
                                />
                              </View>
                              <View
                                onLayout={[Function]}
                                style={null}
                              />
                              <View
                                onLayout={[Function]}
                                style={null}
                              >
                                <View
                                  accessible={true}
                                  collapsable={false}
                                  focusable={true}
                                  onClick={[Function]}
                                  onResponderGrant={[Function]}
                                  onResponderMove={[Function]}
                                  onResponderRelease={[Function]}
                                  onResponderTerminate={[Function]}
                                  onResponderTerminationRequest={[Function]}
                                  onStartShouldSetResponder={[Function]}
                                  style={
                                    Object {
                                      "alignItems": "center",
                                      "alignSelf": "stretch",
                                      "flexDirection": "row",
                                      "minHeight": 50,
                                      "opacity": 1,
                                    }
                                  }
                                  testID="PressableListItemContainer"
                                >
                                  <SvgMock
                                    height={58}
                                    testID="LeftItem"
                                    width={58}
                                  />
                                  <Text
                                    allowFontScaling={false}
                                    marginLeft={true}
                                    style={
                                      Array [
                                        Object {
                                          "flex": 1,
                                          "fontFamily": "Poppins-SemiBold",
                                          "fontSize": 25.5,
                                          "lineHeight": 35.5,
                                          "marginLeft": 29,
                                        },
                                        null,
                                      ]
                                    }
                                    testID="ItemTitle"
                                  >
                                    Lock and unlock card
                                  </Text>
                                  <SvgMock
                                    height={45.5}
                                    testID="RightItem"
                                    width={45.5}
                                  />
                                </View>
                                <View
                                  style={
                                    Array [
                                      Object {
                                        "borderColor": "#F2F3F4",
                                        "borderWidth": 1,
                                        "height": 1,
                                        "width": "100%",
                                      },
                                      undefined,
                                    ]
                                  }
                                />
                              </View>
                              <View
                                onLayout={[Function]}
                                style={null}
                              >
                                <View
                                  accessible={true}
                                  collapsable={false}
                                  focusable={true}
                                  onClick={[Function]}
                                  onResponderGrant={[Function]}
                                  onResponderMove={[Function]}
                                  onResponderRelease={[Function]}
                                  onResponderTerminate={[Function]}
                                  onResponderTerminationRequest={[Function]}
                                  onStartShouldSetResponder={[Function]}
                                  style={
                                    Object {
                                      "alignItems": "center",
                                      "alignSelf": "stretch",
                                      "flexDirection": "row",
                                      "minHeight": 50,
                                      "opacity": 1,
                                    }
                                  }
                                  testID="PressableListItemContainer"
                                >
                                  <SvgMock
                                    height={58}
                                    testID="LeftItem"
                                    width={58}
                                  />
                                  <Text
                                    allowFontScaling={false}
                                    marginLeft={true}
                                    style={
                                      Array [
                                        Object {
                                          "flex": 1,
                                          "fontFamily": "Poppins-SemiBold",
                                          "fontSize": 25.5,
                                          "lineHeight": 35.5,
                                          "marginLeft": 29,
                                        },
                                        null,
                                      ]
                                    }
                                    testID="ItemTitle"
                                  >
                                    Replace card
                                  </Text>
                                  <SvgMock
                                    height={45.5}
                                    testID="RightItem"
                                    width={45.5}
                                  />
                                </View>
                                <View
                                  style={
                                    Array [
                                      Object {
                                        "borderColor": "#F2F3F4",
                                        "borderWidth": 1,
                                        "height": 1,
                                        "width": "100%",
                                      },
                                      undefined,
                                    ]
                                  }
                                />
                              </View>
                              <View
                                onLayout={[Function]}
                                style={null}
                              >
                                <View
                                  accessible={true}
                                  collapsable={false}
                                  focusable={true}
                                  onClick={[Function]}
                                  onResponderGrant={[Function]}
                                  onResponderMove={[Function]}
                                  onResponderRelease={[Function]}
                                  onResponderTerminate={[Function]}
                                  onResponderTerminationRequest={[Function]}
                                  onStartShouldSetResponder={[Function]}
                                  style={
                                    Object {
                                      "alignItems": "center",
                                      "alignSelf": "stretch",
                                      "flexDirection": "row",
                                      "minHeight": 50,
                                      "opacity": 1,
                                    }
                                  }
                                  testID="PressableListItemContainer"
                                >
                                  <SvgMock
                                    height={58}
                                    testID="LeftItem"
                                    width={58}
                                  />
                                  <Text
                                    allowFontScaling={false}
                                    marginLeft={true}
                                    style={
                                      Array [
                                        Object {
                                          "flex": 1,
                                          "fontFamily": "Poppins-SemiBold",
                                          "fontSize": 25.5,
                                          "lineHeight": 35.5,
                                          "marginLeft": 29,
                                        },
                                        null,
                                      ]
                                    }
                                    testID="ItemTitle"
                                  >
                                    Find ATM
                                  </Text>
                                  <SvgMock
                                    height={45.5}
                                    testID="RightItem"
                                    width={45.5}
                                  />
                                </View>
                                <View
                                  style={
                                    Array [
                                      Object {
                                        "borderColor": "#F2F3F4",
                                        "borderWidth": 1,
                                        "height": 1,
                                        "width": "100%",
                                      },
                                      undefined,
                                    ]
                                  }
                                />
                              </View>
                              <View
                                onLayout={[Function]}
                                style={null}
                              >
                                <View
                                  accessible={true}
                                  collapsable={false}
                                  focusable={true}
                                  onClick={[Function]}
                                  onResponderGrant={[Function]}
                                  onResponderMove={[Function]}
                                  onResponderRelease={[Function]}
                                  onResponderTerminate={[Function]}
                                  onResponderTerminationRequest={[Function]}
                                  onStartShouldSetResponder={[Function]}
                                  style={
                                    Object {
                                      "alignItems": "center",
                                      "alignSelf": "stretch",
                                      "flexDirection": "row",
                                      "minHeight": 50,
                                      "opacity": 1,
                                    }
                                  }
                                  testID="PressableListItemContainer"
                                >
                                  <SvgMock
                                    height={58}
                                    testID="LeftItem"
                                    width={58}
                                  />
                                  <Text
                                    allowFontScaling={false}
                                    marginLeft={true}
                                    style={
                                      Array [
                                        Object {
                                          "flex": 1,
                                          "fontFamily": "Poppins-SemiBold",
                                          "fontSize": 25.5,
                                          "lineHeight": 35.5,
                                          "marginLeft": 29,
                                        },
                                        null,
                                      ]
                                    }
                                    testID="ItemTitle"
                                  >
                                    Find cash deposit location
                                  </Text>
                                  <SvgMock
                                    height={45.5}
                                    testID="RightItem"
                                    width={45.5}
                                  />
                                </View>
                              </View>
                            </View>
                          </RCTScrollView>
                          <Modal
                            animationType="none"
                            hardwareAccelerated={false}
                            onRequestClose={[Function]}
                            supportedOrientations={
                              Array [
                                "portrait",
                                "portrait-upside-down",
                                "landscape",
                                "landscape-left",
                                "landscape-right",
                              ]
                            }
                            transparent={true}
                            visible={false}
                          />
                        </View>
                      </RCTSafeAreaView>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenContainer>
      <View
        collapsable={false}
        pointerEvents="box-none"
        style={
          Object {
            "height": 44,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 1,
          }
        }
      />
    </View>
  </RNCSafeAreaProvider>
</View>
`;
